roles/jenkins_job_launcher/tasks/main.yml

```
@@ -0,0 +1,39 @@
+---
+# tasks file for jenkins-job-launcher
+    
+- name: trigger jenkins job and wait for it to succeed
+  ansible.builtin.uri:
+    url: "{{job_url}}/buildWithParameters/"
+    method: POST
+    user: "{{ username }}"
+    password: "{{ token }}"
+    force_basic_auth: true
+    status_code: [200,201]
+    body: "{{ job_params }}"
+    headers:
+      Content-Type: application/x-www-form-urlencoded
+  register: queue_build_number 
+
+- name: get result status code
+  ansible.builtin.uri:
+    url: '{{ queue_build_number.location }}/api/json'
+    method: GET
+    user: "{{ username }}"
+    password: "{{ token }}"
+  register: jenkins_build_number
+
+- name: get jenkins status code
+  ansible.builtin.uri:
+    url: "{{ jenkins_build_number.json.executable.url }}/api/json"
+    method: GET
+    user: "{{ username }}"
+    password: "{{ token }}"
+  register: in_progress
+  until: in_progress.json.inProgress == "false"
```

if there's intermittent issue to get the Json data task will fail and won't retry therefore better to add default value:
```suggestion
  until: in_progress.json.inProgress|default("true") == "false"
```